compiler/src/main/scala/org/scalajs/core/compiler/JSEncoding.scala

```diff
@@ -103,6 +103,8 @@ trait JSEncoding extends SubComponent { self: GenJSCode =>
     val idSuffix =
       if (sym.isPrivate || allRefClasses.contains(sym.owner))
         sym.owner.ancestors.count(!_.isTraitOrInterface).toString
+      else if (sym.isOuterField)
+        sym.owner.name
```

Comment:
A possible variant would be search for the clashes in the compiler. Then use suffixes `"f"`, `"f1"`, `"f2"`, ... for each version on `$outer`.
In case of no clash we would retain the same names. Otherwise I assume that we don't care too much as the code could not be compiled previously.